graphql-server/src/schemas/neo4j.ts

```diff
@@ -51,41 +58,85 @@ export async function buildNeo4jDbSchema() {
   return neo4jDbSchema;
 }
 
-function buildResolvers(ogm: OGM, apolloClient: typeof ApolloClient) {
+function buildResolvers(
+  ogm: OGM,
+  apolloClient: ApolloClient<NormalizedCacheObject>
+) {
   return {
     Mutation: {
-      async updateSamples(_source: any, { where, update }: any) {
+      async updateSamples(
+        _source: any,
+        { where, update }: { where: SampleWhere; update: SampleUpdateInput }
+      ) {
+        // Grab data passed in from the frontend
         const primaryId =
-          where.hasMetadataSampleMetadataConnection_SOME.node.primaryId;
+          where.hasMetadataSampleMetadataConnection_SOME!.node!.primaryId!;
+
+        const sampleKeyForUpdate = Object.keys(
+          update
+        )[0] as keyof SampleUpdateInput;
 
+        const changedFields = (
+          update[sampleKeyForUpdate] as Array<
+            | SampleHasMetadataSampleMetadataUpdateFieldInput
+            | SampleHasTempoTemposUpdateFieldInput
+          >
+        )[0].update!.node!;
+
+        // Get sample manifest from SMILE API /sampleById and update fields that were changed
         const sampleManifest = await request(
           props.smile_sample_endpoint + primaryId,
           {
             json: true,
           }
         );
 
-        const sampleKeyForUpdate = Object.keys(update)[0];
-
-        const changesByPrimaryId = update[sampleKeyForUpdate][0].update.node;
+        Object.keys(changedFields).forEach((changedField) => {
+          const key = changedField as keyof SampleMetadataUpdateInput;
+          sampleManifest[key] =
+            changedFields[key as keyof typeof changedFields];
+        });
+
+        // Get the sample data from the database and update the fields that were changed
+        const updatedSamples: ApolloQueryResult<UpdateSamplesMutationResponse> =
+          await apolloClient.query({
+            query: SamplesDocument,
+            variables: {
+              where: {
+                smileSampleId: sampleManifest.smileSampleId,
+              },
+              hasMetadataSampleMetadataOptions2: {
+                sort: [{ importDate: SortDirection.Desc }],
+                limit: 1,
+              },
+            },
+          });
+
+        Object.keys(changedFields).forEach((key) => {
+          const sample = updatedSamples.data.samples[0][sampleKeyForUpdate] as
+            | SampleMetadata[]
+            | Tempo[];
+          if (Array.isArray(sample) && sample.length > 0) {
+            sample[0][key as keyof typeof sample[0]] =
+              changedFields[key as keyof typeof changedFields];
+          }
+        });
```

Comment:
This code block takes the queried sample data (see `query Samples` in `operations.graphql` for the shape of this data), updates it with changes from users, and then returns it to the frontend. This is one of the requirements for optimistic response and was previously missing for the Cohort Sample update flow.